cygnus-ngsi/src/test/java/com/telefonica/iot/cygnus/sinks/NGSIHDFSSinkTest.java

```diff
@@ -1196,5 +1212,352 @@ private Context createContext(String backendImpl, String backendMaxConns, String
+        NGSIHDFSSink.JSONColumnAggregator columnAggregator = ngsihdfsSink.new JSONColumnAggregator();
+        NotifyContextRequest.ContextElement contextElement = createContextElementForNativeTypes();
+        NGSIEvent ngsiEvent = new NGSIEvent(headers, contextElement.toString().getBytes(), contextElement, null);
+        columnAggregator.initialize(ngsiEvent);
+        columnAggregator.aggregate(ngsiEvent);
+        System.out.println(getTestTraceHead("[NGSIHDFSSink.testNativeTypeColumnJson]")
+                + "-  OK");
+        String correctAggregation = "{\"recvTime\":\"2016-04-20T07:19:55.801Z\",\"fiwareServicePath\":\"somePath\",\"entityId\":\"someId\",\"entityType\":\"someType\", \"someNumber\":\"2\", \"someNumber_md\":[], \"somneBoolean\":\"true\", \"somneBoolean_md\":[], \"someDate\":\"2016-09-21T01:23:00.00Z\", \"someDate_md\":[], \"someGeoJson\":\"{\"type\": \"Point\",\"coordinates\": [-0.036177,39.986159]}\", \"someGeoJson_md\":[], \"someJson\":\"{\"String\": \"string\"}\", \"someJson_md\":[], \"someString\":\"foo\", \"someString_md\":[], \"someString2\":\"\", \"someString2_md\":[]}";
+        if (columnAggregator.getAggregation().equals(correctAggregation)) {
+            assertTrue(true);
+        } else {
+            fail();
+            System.out.println(getTestTraceHead("[NGSIHDFSSink.testNativeTypeColumnJson]")
+                    + "-  FAIL");
+        }
+
+    }//testNativeTypeColumnJson
+
+    @Test
+    public void testNativeTypeColumnCsv() throws CygnusBadConfiguration, CygnusBadContextData {
+        String attr_native_types = "true"; // default
+        NGSIHDFSSink ngsihdfsSink= new NGSIHDFSSink();
+        ngsihdfsSink.configure(createContextforNativeTypes(null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, attr_native_types));
+        // Create a NGSIEvent
+        String timestamp = "1461136795801";
+        String correlatorId = "123456789";
+        String transactionId = "123456789";
+        String originalService = "someService";
+        String originalServicePath = "somePath";
+        String mappedService = "newService";
+        String mappedServicePath = "newPath";
+        Map<String, String> headers = new HashMap<>();
+        headers.put(NGSIConstants.FLUME_HEADER_TIMESTAMP, timestamp);
+        headers.put(CommonConstants.HEADER_CORRELATOR_ID, correlatorId);
+        headers.put(NGSIConstants.FLUME_HEADER_TRANSACTION_ID, transactionId);
+        headers.put(CommonConstants.HEADER_FIWARE_SERVICE, originalService);
+        headers.put(CommonConstants.HEADER_FIWARE_SERVICE_PATH, originalServicePath);
+        headers.put(NGSIConstants.FLUME_HEADER_MAPPED_SERVICE, mappedService);
+        headers.put(NGSIConstants.FLUME_HEADER_MAPPED_SERVICE_PATH, mappedServicePath);
+        //NGSICKANSink.ColumnAggregator columnAggregator = ngsickanSink.new ColumnAggregator(false, false, false, false);
```

Comment:
Leftover comment? Should be removed?

Reply:
Fixed in [9e61357](https://github.com/telefonicaid/fiware-cygnus/pull/1799/commits/9e61357a49559bffebc01a71389c87be49ebd72c)